zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,145 @@
+that will process spaces and pages every 5 minutes is:
+*/5 * * * * /usr/local/share/zulip/integrations/confluence/confluence [options]
+
+== Setting up Confluence authentications ==
+
+Run this on a personal or trusted machine, because your API key is
+visible to local users through the command line or config file.
+
+This bot uses OAuth to authenticate with Confluence. Please create a
+~/.zulip_confluencerc with the following contents:
+
+[Confluence_api]
+username =
+consumer_key =
+domain_Url =
+
+
+In order to obtain a consumer key & secret, you must register a
+new application under your Confluence account:
+
+1 Log in to https://id.atlassian.com.
+2 Click API tokens, then Create API token.
+3 Use Copy to clipboard, and paste the token to your script
+  or elsewhere:
+
+"""
+parser = zulip.add_default_arguments(argparse.ArgumentParser("Fetch space and page from Confluence."))
+parser.add_argument('--instructions',
+                    action='store_true',
+                    help='Show instructions for the confluence bot setup and exit'
+                    )
+opts = parser.parse_args()
+if opts.instructions:
+    print(INSTRUCTIONS)
+    sys.exit()
+if opts.zulip_config_file:
+    zuliprc_path = opts.zulip_config_file
+
+try:
+    config = ConfigParser()
+    config.read(CONFIGFILE)
+    username = config.get('Confluence_api', 'username')
+    consumer_key = config.get('Confluence_api', 'consumer_key')
+    domain_Url = config.get('Confluence_api', 'domain_Url')
+except Exception:
+    parser.error("Please provide a ~/.zulip_confluencerc")
+
+local_file = "zulip.txt"
+f = open(local_file, "w+")
+f.write('0')
+f.close()
+
+def confluence_api() -> None:
+
+    params = ('type', 'page')
+    base_url = 'https://{}/wiki/rest/api/content'.format(domain_Url)
+    auth = (username, consumer_key)
+    response = requests.get(base_url, params=params, auth=auth)
+    res = response.json()
+    # length of json data
+    length_api = len(res['results'])
+    # Fetch value from database models
+    f = open(local_file, "r")
+    length_confluence_data = int(f.read())
+    # compare the lenth of json from database to the new
+    # length of json which is fetch from api
+    # if length of fetch json from api  is greater than
+    # save length then we call zulip api
+    if (length_api > length_confluence_data):
+        test1_json = (res['results'][length_api-1]['_links']['webui'])
+        title = res['results'][length_api-1]['title']
+        get_spaces_and_pages = spacepage(test1_json)
+        format_page_and_string = "New {} is created with title {}".format(get_spaces_and_pages, title)
+        client = zulip.Client(config_file = zuliprc_path)
+        request = {
+            "type": "stream",
+            "to": "confluence",
+            "subject": "Page",
+            "content": format_page_and_string
+        }
+        result = client.send_message(request)
+        f = open(local_file, "w")
```

Comment:
I don't think you need to write this to a local file. If you'd like to add some sort of logging feature, I would do that after we have the base implementation working.

Reply:
@eeshangarg I'm not too familiar with the Confluence API, but what I think he is trying to do here is compare the length of the response received from the current request to the length of the response received from the previous request (which was saved in the local file). Since I'm not too familiar with the Confluence API, I'm not sure if thats the best way to do it.

Reply:
Yeah, but response length isn't really a good indicator of anything, so we should probably avoid this anyway.

Reply:
Agreed; also that approach doesn't work if the Zulip deployment has multiple servers, in any case.

Reply:
@eeshangarg I read up a bit. I think [this](https://developer.atlassian.com/server/confluence/advanced-searching-using-cql/) can be useful to constantly poll for new changes without having to create a local file. Also, I'm not sure why I can't mention paras112 anymore.

Reply:
@paras112 If you are no longer working on this, I can pick it up from here and try to finish it.

Reply:
@mayankmadan Actually I am little bit busy in my sem exam.You could work on this issue and I think Advance Search API work fine here .Thanks